discovery-provider/plugins/network-monitoring/src/content/index.ts

```diff
@@ -76,7 +82,8 @@ export const indexContent = async (run_id: number) => {
 
     await Promise.all(
         // for each content node...
-        content_nodes.map((cnode, _) => {
+        content_nodes.map(async (cnode, _) => {
+
             // const image_cid_count: string = image_to_content_nodes[i]?.cid_count || '0'
             return new Promise<void>(async resolve => {
```

Comment:
can't this be simplified to
```
await Promise.all([
                // check user clock value
                checkUsers(run_id, cnode.spid, cnode.endpoint),

                // check user cids
                // checkCIDS(run_id, cnode.spid, cnode.endpoint, parseInt(cnode.cid_count), parseInt(image_cid_count)),
            ])
```
i.e. remove the `return new Promise` stuff

Reply:
Unfortunately not, what that `map()` and `return new Promise` do, is basically do what you're trying to do above, but **for every content node**

Reply:
yeh i know, i meant doing this

```
await Promise.all(
    content_nodes.map(async (cnode, _) => {
        await Promise.all([
            // check user clock value
            checkUsers(run_id, cnode.spid, cnode.endpoint),

            // check user cids
            // checkCIDS(...)
        ])
    })
)
```

is this not functionally the same?